spec/models/manageiq/providers/amazon/cloud_manager_spec.rb

```diff
@@ -14,6 +14,12 @@
       described_class.raw_connect('access_key', 'secret_access_key', :EC2, 'region', 'uri', true)
     end
 
+    it "validates credentials with an optional uri endpoint" do
+      expect(described_class).to receive(:validate_connection)
+      uri = URI.parse('https://apigateway.us-east-1.amazonaws.com')
+      described_class.raw_connect('access_key', 'secret_access_key', :EC2, 'region', 'uri', true, uri)
```

Comment:
the `uri` parameter is listed twice, can you make the first one `proxy_uri` otherwise it is confusing. Or make the second one `endpoint_uri` if thats easier.